www/js/history-widget.js

```diff
@@ -117,38 +117,47 @@ define([
 		loadHandler : function(e){
 			var label = e.target.id;
 			var session_id = label.slice(4,label.indexOf('_label'));
-			//before attaching the session id or x parameter
-			//verify if it is undefined , then add form property
-			//else just modify the value with current session id depending on history item opted
-			var form = document.forms[formID];
-			if(form.x === undefined){
-				addFormProperty("x",session_id);
-			}
-			else{
-				form.x.value = session_id;
-			}
-			submit();		
+			console.log("session_id",session_id);
+			//history_form has already been created and attached in initial handlers
+			var form = document.forms["history_form"];
+			//update the session id parameter based on the particular session in history
+			//to be loaded identified by session_id
+			form.x.value = session_id;
```

Comment:
yes, for all the sessions which is not current session, button will have a handler attached to it (we push "x" element each time and fill its value with session id the entry corresponds to, this also answers the third comment).